### ISSUE TITLE
Seems to be broke in Ancestor league

### ISSUE THREAD

**Issue by ThirteenEra** [Mtb-3/Weightless#2]
Unfortunately seems to not work in the current league. I was trying to roll boots with Tailwind, and it rolled past them (Despite the frame around it lighting up). Repeated later with onyx amulet when i was trying to hit "gems" (any + gem skill), and it rolled past two of them.

Was working fine last league so perhaps something changed?

(Im using your 4k version btw)

**Comment by Mtb-3** [comment]
Hey, sorry to hear that its not working. I haven't changed anything on my end. I am using the standard resolution version since the beginning of the league and its been fine for me. I'm not sure what it could be but I will re-upload a rescaled version for the 4k.

**Comment by ThirteenEra** [comment]
Its strange because i changed nothing since last league, still using the same file i downloaded previously. And yet it seems to have been broken.

Ill try your re-upload to see if that works

**Comment by ThirteenEra** [comment]
https://github.com/Mtb-3/Weightless/assets/111650549/f24c0d1d-c06d-4574-acdb-bdf781ab230f

**Comment by ThirteenEra** [comment]
Downloaded your new version. Still same issue.
Its strange because nothing in my setup changed since last league.

EDIT: Jeweller's orb rolling works. It stops correctly after item is 6socketed.
EDIT2: Fusing orb rolling works too. Stopped correctly after 6link.

Seems like just mod rolling is broken. I have tried rings, amulets, bows and body armor, with different mods, and it keeps rolling past every time.

**Comment by Mtb-3** [comment]
Do you have python? Maybe I'm off with how I scaled it. I don't have a 4k resolution setup for testing. The way the program works is that it looks for the RGB values of the golden/yellow border within certain dimensions in the currency tab.

**Comment by Mtb-3** [comment]
I am going to upload a .Py file which will screenshot the location of what is supposed to be the item window in the currency tab. If you could link the image I should be able to correct it. 

I can also change it to an .exe if necessary

**Comment by ThirteenEra** [comment]
Exe would be better, i dont have programming setup on my machine

Its just strange because it worked in last league. I used it a lot. So maybe something changed in the backend that affected 4k more than other modes?

**Comment by Mtb-3** [comment]
I uploaded the file. It is a bit slow on startup but it will prompt you to choose a folder location for the output .png file. The name of the file is "scrnshot.png". 

It's really weird why its not working for you. If the window is scaled to the correct dimensions which I think it is. The only other reason I can think of is there is a color difference (RGB values) between now and before.

**Comment by ThirteenEra** [comment]
![scrnshot](https://github.com/Mtb-3/Weightless/assets/111650549/902d10d0-fcb9-4c5d-9ba2-60bb2f455d59)

**Comment by ThirteenEra** [comment]
As mentioned before, i have not changed anything in my setup. Still using same game (didnt reinstall it or anything), same monitor, same gpu, same ingame settings etc. Is it possible that the new engine patch stuff that came with Ancestor changed something?

If you are looking for a specific pixel to be a specific color, maybe we can try to instead look for that pixel changing color? Instead of "it has to be X to trigger" change it to "if it changes to anything other than what it currently is, trigger"? Though im just speaking as a layman here

**Comment by Mtb-3** [comment]
I think you are right that there was a change with the higher resolutions this league. I can't think of anything else but its possible there is something I'm missing. 

I am uploading another test which follows the logic that you proposed. If this fails then it confirms that there is a difference between the colour of your system and mine. I have a final solution that will require a bit more coding but I think it will be possible.

**Comment by ThirteenEra** [comment]
4k final test - does not work. It rolls once and stops, regardless of outcome. The program then auto quits.
Also seems like using the new program makes my ALT key stuck - i have to press alt/ctrl/shift a bunch of times for my system to "unstick" it.

I do have HDR monitor - which could be a difference in color between our systems - however i also had this exact same monitor last league as well, with exact same settings. Still, could be a reason.

**Comment by Mtb-3** [comment]
Alright I am pretty sure that confirms it. Haha thanks for doing all the testing. I think I have the solution just finishing the code now. This version will take a screenshot of your resolution as opposed to using mine. This is probably how I should've coded it to begin with.

**Comment by ThirteenEra** [comment]
I am happy to say that new version works! Ive tried it on a few different items and mods, both in standard and league, and it works great! Thank you!

**Comment by Mtb-3** [comment]
Awesome! Glad to hear it. Thanks again for doing all the testing. As I said a while back I am working on something else but its a bigger project. Would you be interested in beta testing it when its somewhat ready?

**Comment by ThirteenEra** [comment]
Yeah for sure! If its even half as useful as this little tool you have here, id love to assist. POE is a great game but qol in many cases is *severely* lacking
